src/edu/washington/escience/myria/io/UriSource.java

```diff
@@ -44,19 +44,26 @@
    * {@link InputStream}.
    *
    * @param uri the Uniform Resource Indicator (URI) of the data source.
+   * @throws URISyntaxException
    */
   @JsonCreator
-  public UriSource(@JsonProperty(value = "uri", required = true) final String uri) {
-    this.uri = Objects.requireNonNull(uri, "Parameter uri to UriSource may not be null");
+  public UriSource(@JsonProperty(value = "uri", required = true) final String uri) throws URISyntaxException {
+    URI parsedUri = URI.create(Objects.requireNonNull(uri, "Parameter uri to UriSource may not be null"));
```

Comment:
Rather than round-tripping this parsed URI, we could introduce a `transient final` instance variable `parsedUri` which would be set by the constructor and used by `getInputStream()`.